In [181]:
import pandas as pd 
import numpy as np
import os 
from pandasql import sqldf 
import csv
import matplotlib
# Read in CSV files 
Louisville_crime_2020 = pd.read_csv('../Capstone-project---Louisville-crime-/raw_data/Crime_Data_2020.csv')
Louisville_crime_2021 = pd.read_csv('../Capstone-project---Louisville-crime-/raw_data/Crime_Data_2021.csv')
Louisville_crime_2022 = pd.read_csv('../Capstone-project---Louisville-crime-/raw_data/Crime_Data_2022.csv')
#print(Louisville_crime_2022.to_string())

C:\Users\TechLouisville\AppData\Local\Temp\ipykernel_18204\560582393.py:8: DtypeWarning: Columns (13) have mixed types. Specify dtype option on import or set low_memory=False.
  Louisville_crime_2020 = pd.read_csv('../Capstone-project---Louisville-crime-/raw_data/Crime_Data_2020.csv')
C:\Users\TechLouisville\AppData\Local\Temp\ipykernel_18204\560582393.py:9: DtypeWarning: Columns (3) have mixed types. Specify dtype option on import or set low_memory=False.
  Louisville_crime_2021 = pd.read_csv('../Capstone-project---Louisville-crime-/raw_data/Crime_Data_2021.csv')
C:\Users\TechLouisville\AppData\Local\Temp\ipykernel_18204\560582393.py:10: DtypeWarning: Columns (3,14) have mixed types. Specify dtype option on import or set low_memory=False.
  Louisville_crime_2022 = pd.read_csv('../Capstone-project---Louisville-crime-/raw_data/Crime_Data_2022.csv')


In [182]:
# Droping columns by creating a new dataframe
Louisville_crime_2020.columns 
LC20 = Louisville_crime_2020[['DATE_REPORTED','DATE_OCCURED', 'CRIME_TYPE', 
'UOR_DESC', 'ATT_COMP', 'LMPD_DIVISION', 'PREMISE_TYPE', 'ZIP_CODE']].copy()

LC21 = Louisville_crime_2021[['DATE_REPORTED','DATE_OCCURED', 'CRIME_TYPE', 
'UOR_DESC', 'ATT_COMP', 'LMPD_DIVISION', 'PREMISE_TYPE', 'ZIP_CODE']].copy()

LC22 = Louisville_crime_2022[['DATE_REPORTED','DATE_OCCURED', 'CRIME_TYPE', 
'UOR_DESC', 'ATT_COMP', 'LMPD_DIVISION', 'PREMISE_TYPE', 'ZIP_CODE']].copy()
#print(LC22.to_string())

I renamed all the columns on two of the datasets so that I would know what year each column belonged to when I merged them together. I realized that the way I thought of merging the columns was not going to work, so I commented the cell out. 

In [183]:
# renaming columns 
#LC21 = LC21.rename(columns={'DATE_REPORTED': 'DATE_REPORTED21', 'DATE_OCCURED' : 'DATE_OCCURED21', 'CRIME_TYPE' : 'CRIME_TYPE21', 
#'UOR_DESC' : 'UOR_DESC21', 'ATT_COMP' : 'ATT_COMP21',  'LMPD_DIVISION' : 'LMPD_DIVISION21',
#'ZIP_CODE' : 'ZIP_CODE21'})
#LC22 = LC22.rename(columns={'DATE_REPORTED': 'DATE_REPORTED22', 'DATE_OCCURED' : 'DATE_OCCURED22', 'CRIME_TYPE' : 'CRIME_TYPE22', 
#'UOR_DESC' : 'UOR_DESC22', 'ATT_COMP' : 'ATT_COMP22',  'LMPD_DIVISION' : 'LMPD_DIVISION22', 
#'ZIP_CODE' : 'ZIP_CODE22'})
#LC20.to_csv('../Capstone-project---Louisville-crime-/new_data/LC20.csv')
#LC21.to_csv('../Capstone-project---Louisville-crime-/new_data/LC21.csv')
#LC22.to_csv('../Capstone-project---Louisville-crime-/new_data/LC22.csv')
#print(LC20.head(10).to_string())
#print(LC21.head(10).to_string())
#print(LC22.tail(10).to_string())

In [184]:
# Merge datasets 
clean_data = pd.concat([LC20, LC21, LC22], axis=0)
print(clean_data.head(50).to_string())
#clean_data.to_csv('../Capstone-project---Louisville-crime-/new_data/clean_data.csv')

          DATE_REPORTED         DATE_OCCURED                CRIME_TYPE                                                   UOR_DESC   ATT_COMP LMPD_DIVISION                                   PREMISE_TYPE ZIP_CODE
0   2020/01/01 13:50:00  2019/12/31 03:30:00                     FRAUD  FRAUD USE OF CREDIT CARD AFTER REPT LOST/STOLEN U/$10,000  COMPLETED  4TH DIVISION  SERVICE / GAS STATION                            40215
1   2020/01/02 10:45:00  2019/12/31 18:00:00             THEFT/LARCENY               TBUT OR DISP ALL OTHERS > $500 BUT < $10,000  COMPLETED  6TH DIVISION  NON-ATTACHED RESD GARAGE/SHED/BULD               40218
2   2020/01/01 00:26:00  2019/12/31 23:30:00    VEHICLE BREAK-IN/THEFT     TBUT OR DISP CONTENTS FROM VEH $500 OR > BUT < $10,000  COMPLETED  1ST DIVISION  HIGHWAY / ROAD / ALLEY                           40203
3   2020/01/01 23:57:00  2020/01/01 03:00:00                 VANDALISM                               CRIMINAL MISCHIEF-2ND DEGREE  COMPLETED  4TH DIVISION  

In [ ]:
clean_data.shape
clean_data.describe
clean_data.groupby('ZIP_CODE').CRIME_TYPE.value_counts()
clean_data["ZIP_CODE"].unique()
clean_data["CRIME_TYPE"].unique()
clean_data["CRIME_TYPE"].value_counts()
clean_data["ZIP_CODE"].value_counts()
clean_data["ATT_COMP"].value_counts()

In [190]:
clean_data["CRIME_TYPE"].value_counts()

CRIME_TYPE
ASSAULT                     49311
THEFT/LARCENY               34247
OTHER                       30293
VEHICLE BREAK-IN/THEFT      18563
VANDALISM                   18396
DRUGS/ALCOHOL VIOLATIONS    14785
BURGLARY                    13609
MOTOR VEHICLE THEFT         13601
FRAUD                       10853
ROBBERY                      3720
WEAPONS                      2831
SEX CRIMES                   1593
DISTURBING THE PEACE         1038
HOMICIDE                      527
ARSON                         495
DUI                             4
Name: count, dtype: int64

 In querying the data I found that most of the crime of the three years occurred in the 40214 neighborhood. That zip code in the south Louisville area. The second highest was 40203, which is the downtown Louisville area. The crime that occurs the most was assault There were 49,311 in the last three years. The second highest is theft/larceny. 